### ISSUE TITLE
Mention creating a new scope when resolving the scoped IOptionsMonitor in Main.cs

### ISSUE THREAD

**Issue by Procee** [dotnet/AspNetCore.Docs#27519]
In the last section of the page is mentioned that you can resolve the service as follows in Main.cs:

var app = builder.Build();

var option1 = app.Services.GetRequiredService<IOptionsMonitor<MyOptions>>()
    .CurrentValue.Option1;

But I ran into an issue where I tried to resolve an IOptionsSnapshot when configuring an ApiClient which required me to create a new IServiceScope. 

Maybe this is something that can be added to the documentation?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ed39cb0e-aa85-8bba-ffb4-859b56a60723
* Version Independent ID: dc7e3449-650b-d117-5eac-69e5ccb250cb
* Content: [Options pattern in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0#feedback)
* Content Source: [aspnetcore/fundamentals/configuration/options.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/configuration/options.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
See [Dependency injection guidelines](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines)

**Comment by Rick-Anderson** [comment]
@adityamandaleeka  please assign a reviewer.

**Comment by adityamandaleeka** [comment]
@halter73 can you take a quick look at this?

**Comment by halter73** [comment]
> ```csharp
> var app = builder.Build();
> 
> var option1 = app.Services.GetRequiredService<IOptionsMonitor<MyOptions>>()
>     .CurrentValue.Option1;
> ```

The above sample from the document already works as written because `IOptionsMonitor` is a singleton service unlike `IOptionsSnapshot` which is a scoped service and must be resolved from a scope.


@Procee what motivated you to change this to `IOptionsSnapshot`? In the [Use IOptionsSnapshot to read updated data](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0#use-ioptionssnapshot-to-read-updated-data) section of the same document we say the following:

> The difference between `IOptionsMonitor` and `IOptionsSnapshot` is that:
> - `IOptionsMonitor` is a [Singleton service](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#singleton) that retrieves current option values at any time, which is especially useful in singleton dependencies.
> - `IOptionsSnapshot` is a [Scoped service](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection#scoped) and provides a snapshot of the options at the time the `IOptionsSnapshot<T>` object is constructed. Options snapshots are designed for use with transient and scoped dependencies.

If someone has a suggestion for how to clarify this better, I'm open to suggestions. However, I don't think we should demonstrate calling `CreateScope()` to resolve an `IOptionsSnapshot` in Program.cs, because we recommend using `IOptionsMonitor` for this singleton use-case case. `IOptionsSnapshot` is intended to be cached on a per request basis, so it should not be necessary for most `IOptionsSnapshot` consumers to ever call `CreateScope()` themselves.

**Comment by Procee** [comment]
Thanks for your thorough reply.

> @Procee what motivated you to change this to IOptionsSnapshot? In the [Use IOptionsSnapshot to read updated data](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-6.0#use-ioptionssnapshot-to-read-updated-data) section of the same document we say the following:

On every request the IOptions instance needs to be re-evaluated because the application serves multiple tenants, it checks which tenant is used and updates the IOptions instance accordingly with the correct BaseUrl. Which is in turn used by the HttpClient. 

>  `IOptionsSnapshot` is intended to be cached on a per-request basis. Since a scope is created automatically for each request, it should not be necessary for most `IOptionsSnapshot` consumers to ever call `CreateScope()` themselves.

This is indeed the non-singleton use case. The underlying problem is that I was not aware that I had to create a scope to resolve the IOptionsSnapshot and this was causing all kinds of weird bugs. I found this solution in on [Stack Overflow](https://stackoverflow.com/questions/65457511/net-5-ioptionssnapshot-cannot-resolve-scoped-service). And I think the documentation is helped by providing some example on how to do this. 

But if this use case is considered to be to obscure for the documentation I understand. Thank you for your consideration.
